plugins/inputs/prometheus/prometheus.go

```diff
@@ -305,8 +305,10 @@ func (p *Prometheus) Start(a telegraf.Accumulator) error {
 }
 
 func (p *Prometheus) Stop() {
-	p.cancel()
-	p.wg.Wait()
+	if p.MonitorPods {
+		p.cancel()
+		p.wg.Wait()
```

Comment:
Always need to wait, if metrics are added after Stop is called the channel could be closed.

Reply:
Even if `p.wg` is only utilized if `MonitorPods` is set?

Reply:
I believe WaitGroup is always used, otherwise I would say no.  Another option here that @goller mentioned was the null object pattern with cancel normally set to `func() {}`.  This way you could always wait and wouldn't need to line up the booleans.